README.md

```diff
@@ -1,9 +1,65 @@
 # OpenPubkey Reference Implementation
 
-OpenPubkey adds user generated cryptographic signatures to OpenID Connect (OIDC) to enable users to sign messages or artifacts under their OpenID identity. Verifiers can check that these signatures are valid and associated with the signing OpenID identity. OpenPubkey does not add any new trusted parties beyond what is required for OpenID Connect and is fully compatible with existing OpenID Providers (Google, Azure/Microsoft, Okta, OneLogin, Keycloak) without any changes to the OpenID Provider.
-
 This repo contains the current reference implementation of OpenPubkey. The reference implementation is a work in progress.
 
+## What is OpenPubkey?
+
+OpenPubkey adds user or workload generated public keys to OpenID Connect (OIDC) enabling identities to sign messages or artifacts under their OIDC identity. In essence, OpenPubkey is a protocol for getting OpenID Providers (OPs) to bind identities to public keys.
+
+Verifiers can check that these signatures are valid and associated with the signing OpenID identity. OpenPubkey does not add any new trusted parties beyond what is required for OpenID Connect. It is fully compatible with existing OpenID Providers (Google, Azure/Microsoft, Okta, OneLogin, Keycloak) without any changes to the OpenID Provider.
+
+### ID Tokens and PK Tokens
+
+OpenPubkey uses ID Tokens issued by OpenID Providers (OPs) to produce PK Tokens. A PK Token consist of the ID Token and the CIC (Client-Instance Claims) where the CIC contains the user's public key and associated metadata. The ID Token contains a hash of the CIC, so that the values in the CIC including the identities public key are cryptographically bound to the ID Token signed by the OP.
+
+For workload-identity, the hash of the CIC is stored in the ID Token's `aud` claim following the pattern of github actions workload identity. For user-identity, the hash of the CIC is stored in the ID Token's `nonce` claim.
```

Comment:
```suggestion
For workload-identity, the hash of the CIC is stored in the ID Token's `aud` claim following the pattern of GitHub Action's workload identity. For user-identity, the hash of the CIC is stored in the ID Token's `nonce` claim.
```

Reply:
Is it worth adding something to the effect, "Note that the use of the `aud` and `nonce` claims in this manner is completely consistent with the OIDC spec." ?